node_modules/col-whiteboards/lib/api.js

```diff
@@ -1439,38 +1439,25 @@ var exportWhiteboardToAsset = module.exports.exportWhiteboardToAsset = function(
 
     title = title || whiteboard.title;
 
-    // Ensure that all provided categories exist
+    // Ensure that all provided categories exist.
     CategoriesAPI.getCategoriesById(ctx, opts.categories, function(err, categories) {
       if (err) {
         return callback(err);
       }
 
-      AssetsAPI.createWhiteboard(ctx, whiteboard, title, opts, function(err, asset) {
+      // Get an updated image for the whiteboard.
+      generateThumbnail(whiteboard, function(err, whiteboard, dimensions) {
         if (err) {
-          return callback(err);
-        }
-
-        // Pass the asset to the caller
-        callback(null, asset, whiteboard);
-
-        // Asynchronously create a thumbnail for the exported whiteboard asset. If the preview integration
-        // has not been enabled, we can't generate a thumbnail and simply return
-        if (!config.get('previews.enabled')) {
-          return;
+          log.error({'err': err}, 'Error creating a PNG from a newly exported whiteboard');
+          return callback({'code': 500, 'msg': 'The board could not be exported'});
         }
 
-        generateThumbnail(whiteboard, function(err, whiteboard, dimensions) {
+        AssetsAPI.createWhiteboard(ctx, whiteboard, title, opts, function(err, asset) {
           if (err) {
-            return log.error({'err': err}, 'Could not generate a thumbnail for an exported whiteboard');
+            return callback(err);
           }
 
-          if (whiteboard.image_url) {
-            updateAssetPreview(asset.id, whiteboard.image_url, dimensions.width, function(err) {
```

Comment:
Part 2 of the bug: the whiteboards API was not calling `AssetsAPI.updateAssetPreview` (which would have performed properly) but `WhiteboardsAPI.updateAssetPreview` (which does something completely different).

With this refactor, `AssetsAPI.createWhiteboard` (above) now calls `AssetsAPI.updateAssetPreview` as part of its business.